homeworks/Vladimir.Karbivnychiy_VladimirSkarb/11-dependency-injection/car-list/src/app/directives/change-color-field.directive.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Directive,
+  ElementRef,
+  AfterViewInit,
+  Input,
+  HostListener
+} from '@angular/core';
+import {Vehicle} from "../models";
+
+@Directive({selector: '[appChangeColorField]'})
+export class ChangeColorFieldDirective implements AfterViewInit {
+  @Input('appChangeColorField') vehicle: Vehicle | undefined
+
+  constructor(private el: ElementRef) {
+  }
+
+  ngAfterViewInit() {
+    const protocol: string = this.el.nativeElement.children[1].innerHTML;
+    if (protocol === 'http') {
+      this.el.nativeElement.children[0].style.color = 'red';
+      this.el.nativeElement.classList.add('no-link');
+    }
+  }
+
+  @HostListener('click')
+  onMouseClick(): void {
+    window.open(`https://www.google.com/search?q=${this.vehicle?.manufacturer}+${this.vehicle?.model}`, '_blank')
```

Comment:
This is not a good option, because:
1) This is not a native open. When you as the user tries to open a new page, you expect to see where you will be forwarded (at the left bottom corner). Use the `<a>` tag instead

2) Your URL is correct, but it's better to use the native `encodeURI` function.
You can read about it here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI